### ISSUE TITLE
mask r-cnn not support

### ISSUE THREAD

**Issue by ujeanBal** [tensorflow/tfjs#1094]
Hi, everyone. I want to use https://github.com/matterport/Mask_RCNN/ but in browser.
So I:
1. download mask_rcnn_balloon.h5 from matterport releases.
2. in jupyter notebook instead of model.keras_model.save_weights(model_path), i call model.keras_model.save(model_path) - to save whole model (not only weights).
3. convert to model.json (with graphs) through tensorflowjs_converter.

And try to load it in javascript, and get error
ERROR Error: Uncaught (in promise): Error: Unknown layer: Lambda. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().
Error: Unknown layer: Lambda. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().

So i create fake layers, but also i  read in https://keras.io/layers/writing-your-own-keras-layers/
So to create custom keras layers in js i need to implement:

- build(input_shape)
- call(x)
- compute_output_shape(input_shape).

But I dont know details of that layers...(and i dont have experience in python and keras to convert python code to js)
And there are few unknown keras layers : BatchNorm, Lambda, ProposalLayer, PyramidROIAlign.
Please can you add support mask r-cnn in javascript?

**Comment by bileschi** [comment]
Hi @ujeanBal ,

Thanks for your question.
There is already an implementation of the BatchNormalization layer, assuming that is the same as the BatchNorm layer, here :+1: https://js.tensorflow.org/api/0.14.1/#layers.batchNormalization

Lambda layers, as currently implemented in python keras, are layers which run python code directly.  This can not be made to work in JS, because of the lack of a python runtime.

I have not heard of the ProposalLayer or PyramidROIAlign layers, are these custom layers?   Writing a custom layer is an advanced technique, but it is possible.  You can see an example here:

https://github.com/tensorflow/tfjs-examples/tree/master/custom-layer

Does this answer your question?  If not, please feel free to re-open this issue.

Thanks!
--Stan

**Comment by chenqing** [comment]
Error: Unknown layer: Lambda. This may be due to one of the following reasons:
1. The layer is defined in Python, in which case it needs to be ported to TensorFlow.js or your JavaScript code.
2. The custom layer is defined in JavaScript, but is not registered properly with tf.serialization.registerClass().

**Comment by kavikode** [comment]
how can we convert the Mask R-CNN model to Tensorflow.js?

**Comment by oxedom** [comment]
Any updates about this?

**Comment by skabbit** [comment]
still looking for solution
